src/NuGet.Core/NuGet.Frameworks/DualCompatibilityFramework.cs

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using NuGet.Shared;
+
+namespace NuGet.Frameworks
+{
+    /// <summary>
+    /// Represents a framework that behaves as 2 potentially independent frameworks.
+    /// Ex. C++/CLI can support both .NET 5.0 and native.
+    /// This type is immutable.
+    /// </summary>
+    public class DualCompatibilityFramework : NuGetFramework
+    {
+        /// <summary>
+        /// The root framework. Any compatibility checks should be performed against this framework first.
+        /// </summary>
+        public NuGetFramework RootFramework { get; }
+
+        /// <summary>
+        /// The secondary framework. If the root framework compatibility checks fail, then the compat checks should be performed against this framework next.
+        /// </summary>
+        public NuGetFramework SecondaryFramework { get; }
+
+        private int? _hashCode;
+
+        /// <summary>
+        /// Multiple compatbility 
+        /// </summary>
+        /// <param name="framework">Root framework. Never <see langword="null"/>. </param>
+        /// <param name="secondaryFramework">Secondary framework. Never <see langword="null"/>. </param>
+        /// <exception cref="ArgumentNullException">if either <paramref name="framework"/> or <paramref name="secondaryFramework"/> are <see langword="null"/>.</exception>
+        public DualCompatibilityFramework(NuGetFramework framework, NuGetFramework secondaryFramework)
+            : base(ValidateFrameworkArgument(framework))
+        {
+            if (secondaryFramework == null)
+            {
+                throw new ArgumentNullException(nameof(secondaryFramework));
+            }
+
+            SecondaryFramework = secondaryFramework;
```

Comment:
we should call `ValidateFrameworkArgument(secondaryFramework)` also before setting the value just like `ValidateFrameworkArgument(framework)`.

Reply:
That would change the message. 
The static helper exists because the value is getting passed to the base constructor and it's not getting validated in the NuGetFramework consturctor.